repoman/git/depot_operations.py

```diff
@@ -50,24 +50,31 @@ def grab_changesets(self, path, url, changesets):
         logger.debug('Grabbing changesets from %s to %s' % (url, path))
         git_repo = pygit2.Repository(path)
         try:
-            origin = self._set_origin_source(git_repo, url)
-            #this may fail against github
-            result = origin.fetch()
-            logger.debug('GIT Done grabbing changesets (%s)' % (result))
-        except (pygit2.GitError, OSError) as e:
-            logger.exception('Error Grabbing changesets: %s' % e)
-            logger.info("Retrying using process")
-            try:
+            self._set_origin_source(git_repo, url)
```

Comment:
Doing `git fetch origin`, it also fails. These is the output:
```
fatal: Refusing to fetch into current branch refs/heads/integration of non-bare repository
fatal: The remote end hung up unexpectedly
```

The reason for that, I think it is for how the remote refs are managed in method `_set_origin_source`: 
```
origin.set_fetch_refspecs(['+refs/*:refs/*'])
origin.save()
```